erpnext/regional/india/e_invoice/utils.py

```diff
@@ -979,6 +992,30 @@ def generate_irn(self):
 						Contact ERPNext support to resolve the issue."
 					)
 
+			if res.get("info"):
+				info = res.get("info")
+				"""
+    				"info": [
+    				    {
+    				        "InfCd": "EWBERR",
+    				        "Desc": [
+    				            {
+    				                "ErrorCode": "4011",
+    				                "ErrorMessage": "Vehicle number  should be passed in case of transportation mode is Road."
+    				            }
+    				        ]
+    				    }
+    				]
+				"""
+				# right now eway bill errors are shown, we can add more when available using below forloop.
+				for msg in info:
+					if msg.get("InfCd") == "EWBERR":
+						for desc in msg.get("Desc"):
+							frappe.msgprint(
+								_(desc.get("ErrorMessage")),
+								title="E-Way Bill not generated",
+								indicator="orange",
+							)
```

Comment:
This could be extracted into a separate method?

Reply:
I can extract for code readability but other than that there is really no other use for this as of now only case where IRN status is successful and there is an error inside info is when we have an e-way bill error.

Reply:
I will commit separate method soon

Reply:
I have made changes as discuessed.